### PR TITLE
store/tikv: Fix canceling range task may not return error

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

### What problem does this PR solve? 

The RangeTaskRunner has a small problem. If a running task is canceled by the outer ctx, just after finishing sending subtasks to workers but before the workers finishing working, then it's a little chance that the task is not fully finished but the func `RunOnRange` doesn't return error. This may not cause any big problem, but it's better to fix it.

### What is changed and how it works?

When the worker is canceled, set its error status.

### Check List 

Tests 

 - No

Code changes

 - No

Side effects

 - No

Related changes

 - No

### CODE REVIEW

File: store/tikv/range_task.go [hunk 1]
```diff
@@ -241,6 +241,7 @@ func (w *rangeTaskWorker) run(ctx context.Context, cancel context.CancelFunc) {
 	for r := range w.taskCh {
 		select {
 		case <-ctx.Done():
+			w.err = errors.Trace(ctx.Err())
```
Comment (ngaut) [inline store/tikv/range_task.go]: No need to record stack trace.

**Comment by zz-jason** [comment]
how about adding a failpoint test?

**Comment by disksing** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
@MyonKeminta Don't forget to add a failpoint test in another PR.

**Comment by MyonKeminta** [comment]
Ok.

**Pull request merged**